### ISSUE TITLE
Handling Java exceptions

### ISSUE THREAD

**Issue by zonyitoo** [hkuplg/fcore#136]
There is no way to handle/raise exception in F2J, so I cannot even read a file.

Also for #108 .

**Comment by zhiyuanshi** [comment]
Any one working on this?

**Comment by bixuanzju** [comment]
@zhiyuanshi I think no one

**Comment by zhiyuanshi** [comment]
@zonyitoo Could you provide a minimal program that exhibits the issue?
@bixuanzju I assign this to myself now. After I've sorted out the frontend, I'll reassign the backend part to you.

**Comment by bixuanzju** [comment]
@zhiyuanshi sure!

**Comment by zonyitoo** [comment]
first example, the simplest,

```
exception LogicalException String

def head[A] (l : List[A]) : A =
    case l of
        Nil         -> raise (LogicalException "Call `head` on an empty List")
     |  Cons x _    -> x
```

**Comment by zhiyuanshi** [comment]
First we handle Java exceptions and then make our own as Java exceptions. :) So what I need is a F2J program that uses Java file I/O and the resulting Java program should not be able to be compiled by `javac` due to the fact that exception is not caught.

**Comment by zonyitoo** [comment]
@zhiyuanshi So you want a program that cannot be complied? That is easy ;)

```
def readFileToString (name : String) : String =
    (new java.util.Scanner(new java.io.File(name))).useDelimiter("\\Z").next()
```

**Comment by zhiyuanshi** [comment]
@zonyitoo Oh great. I like it.

**Comment by bixuanzju** [comment]
What's the status now?

**Comment by zhiyuanshi** [comment]
@xnning Wondering if you're handling this for us at the moment?

**Comment by xnning** [comment]
@zhiyuanshi Nope. Pull request #168  provides a `error` operation (needed to be polished), which is more related to #108 .
